openstack/baremetalintrospection/v1/introspection/results.go

```diff
@@ -240,6 +247,28 @@ type SystemVendorType struct {
 	SerialNumber string `json:"serial_number"`
 }
 
+// UnmarshalJson interprets an LLDP TLV [key, value] pair as an LldpTlv structure
+func (tlv *LldpTlvType) UnmarshalJSON(data []byte) error {
+	var list []interface{}
+	if err := json.Unmarshal(data, &list); err != nil {
+		return err
+	}
+
+	fieldtype, ok := list[0].(float64)
+	if !ok {
+		return errors.New("LLDP TLV key is not number")
```

Comment:
nit: `fmt.Errorf` instead of `errors.New`. (There are a handful of `errors.New` in Gophercloud, but they should be changed to `fmt.Errorf` at some point.)

The two are virtually identical - it's more just an internal coding style.